cmd/ethrex/l2/command.rs

```diff
@@ -269,142 +284,156 @@ impl Command {
+                        account_updates: state_updates,
+                        storage_updates: accounts_updates,
+                        blocks: vec![],
+                        receipts: vec![],
+                        code_updates: vec![],
+                    };
+
+                    store
+                        .store_block_updates(pseudo_update_batch)
+                        .await
+                        .expect("Error storing trie updates");
+
+                    new_trie = store
+                        .open_state_trie(new_state_root)
+                        .expect("Error opening new state trie");
+
+                    // Get withdrawal hashes
+                    let message_hashes = state_diff
+                        .l1_messages
+                        .iter()
+                        .map(get_l1_message_hash)
+                        .collect();
+
+                    // Get the first block of the batch
+                    let first_block_number = last_block_number + 1;
+
+                    // Build the header of the last block.
+                    // Note that its state_root is the root of new_trie.
+                    let new_block = BlockHeader {
+                        coinbase,
+                        state_root: new_trie.hash().expect("Error committing state"),
+                        ..state_diff.last_header
+                    };
+
+                    // Store last block.
+                    let new_block_hash = new_block.hash();
+                    store
+                        .add_block_header(new_block_hash, new_block.clone())
+                        .await?;
+                    store
+                        .add_block_number(new_block_hash, state_diff.last_header.number)
+                        .await?;
+                    store
+                        .set_canonical_block(state_diff.last_header.number, new_block_hash)
+                        .await?;
+                    println!(
+                        "Stored last block of blob. Block {}. State root {}",
+                        new_block.number, new_block.state_root
+                    );
+
+                    last_block_number = new_block.number;
+
+                    let batch = Batch {
+                        number: batch_number,
+                        first_block: first_block_number,
+                        last_block: new_block.number,
+                        state_root: new_block.state_root,
+                        privileged_transactions_hash: H256::zero(),
+                        message_hashes,
+                        blobs_bundle: BlobsBundle::empty(),
+                        commit_tx: None,
+                        verify_tx: None,
+                    };
+
+                    // Store batch info in L2 storage
+                    rollup_store
+                        .seal_batch(batch)
+                        .await
+                        .expect("Error storing batch");
```

Comment:
this consumes the error reason

Reply:
Improved in [eb4cb38](https://github.com/lambdaclass/ethrex/pull/3826/commits/eb4cb3864cbe789bac984481a83664e7f28ea0a3)